### PR TITLE
Task: tweak docs

### PR DESCRIPTION
**Pull request by oubiwann**
As part of the new-dev onboarding, this updates the docs.

In particular, the following were done:

* Added docs for testing
* Added docs for new static analysis features
* Added missing steps in old docs
* Updated commands for consistency
* Updated a script to be used from either the top-level (as originally documented) or from `dev-system`
* Updated READMEs with long lines to be more easily read in the terminal

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,73 +1,201 @@
...
 #### Building and Running CMR Dev System in a REPL
 
-1. Install Oracle JDBC Jars into your local maven repository following instructions in `oracle-lib/README.md`. The CMR must have these libraries to build but it does not depend on Oracle DB when running locally. It uses a local in memory database by default.
-2. `cd cmr/dev-system`
-3. `./support/setup_local_dev.sh`
+1. Install Oracle JDBC Jars into your local maven repository following
+   instructions in `oracle-lib/README.md`. The CMR must have these libraries to
+   build but it does not depend on Oracle DB when running locally. It uses a
+   local in memory database by default.
+2. `cd cmr`
```
Comment (mschmele) [inline README.md]: Since the repo name changed it should be:

`cd cmr` -> `cd Common-Metadata-Repository`

File: README.md [hunk 2]
```diff
@@ -1,73 +1,201 @@
...
 4. `lein repl`
 5. Once given a Clojure prompt, run `(reset)`
 
+Note that the `reset` action could potentially take a while, not only due to
+the code reloading for a large number of namespaces, but for bootstrapping
+services as well as starting up worker threads.
+
 #### Building and Running CMR Dev System from a Jar
 
 1. `cd cmr/dev-system`
```
Comment (mschmele) [inline README.md]: `cd cmr/dev-system` -> `cd Common-Metadata-Repository/dev-system`

File: README.md [hunk 3]
```diff
@@ -1,73 +1,201 @@
...
-This will build all of the applications, but will put each jar into the appropriate /target directory for each application.
-The command shown in step 3 is an example. For the proper command to start up each application, see the `Applications` section below. Note: Steps 1 and 2 only need to be completed once.
+
+This will build all of the applications, but will put each jar into the
+appropriate /target directory for each application. The command shown in step
+3 is an example. For the proper command to start up each application, see the
+`Applications` section below. Note: Steps 1 and 2 only need to be completed
+once.
 
 1. `cd cmr/dev-system`
```
Comment (mschmele) [inline README.md]: Update repo name here as well

File: README.md (hunk 1 above)
Comment (oubiwann) [inline README.md]: Ah, yes ... what do you think about updated the instructions to use `git checkout <REPO URL> cmr`? 

(That will keep in consistent with the old and also make things easier to type over time :-) )

Comment (mschmele) [inline README.md]: Oh that's a good idea. Less to change too!

Comment (oubiwann) [inline README.md]: Sweet. Will update instructions :-)

Comment (oubiwann) [inline README.md]: Done.

File: README.md (hunk 2 above)
Comment (oubiwann) [inline README.md]: Addressed per comments above.

File: README.md (hunk 3 above)
Comment (oubiwann) [inline README.md]: Addressed per comments above.

**Pull request merged**